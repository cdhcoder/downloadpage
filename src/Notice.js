import React, { useState, useEffect } from 'react';
import axios from 'axios';

function Notice() {

    const [notis, setNotis] = useState(null);
    const [loading, setLoading] = useState(false);
    const [error, setError] = useState(null);

    useEffect(() => {

        const fetchNotis = async () => {
            try {
                setError(null);
                setNotis(null);
                
                setLoading(true);
                const response = await axios.get('http://192.168.1.211:8080/api/notice');
                setNotis(response.data.data);
            } catch (e) {
                setError(e);
            }
            setLoading(false);
        };

        fetchNotis();
        return () => setLoading(false);
    }, []);

    if (loading) return <div>Loading...</div>;
    if (error) return <div>An error occurred</div>;
    if (!notis) return null;

    return (
        <div className="card">
            <h2 id="Notice"><font color="#0066ff">&#9483; </font>Notice</h2>
            <table>
                <tbody>
                    {notis.map((noti, index) => (
                        <tr>
                            <td className="row-title" key={index}>
                                <strong>&#x2611; {noti.version}</strong>
                            </td>
                            <td>
                                {noti.date}
                            </td>
                        </tr>
                    ))}
                </tbody>
            </table>
        </div>
    );
}

export default Notice;