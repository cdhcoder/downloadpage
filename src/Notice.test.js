import { render, waitFor } from '@testing-library/react';
import Notice from './Notice';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

describe('<Notice/>', () => {

    const mock = new MockAdapter(axios, { delayResponse: 200 });    //200ms delay

    //dummy response
    mock.onGet('http://192.168.1.211:8080/api/notice').reply(200, {
        result: 1,
        data: [{
            version: 'TmaxOS 21 업데이트',
            date: '2021.03.04'
        }]
    });

    it('matches snapshot', async () => {
        const { container } = render(<Notice/>);
        expect(container).toMatchSnapshot();
    });

    it('calls get Notice API loads notice data properly', async () => {
        const { debug, getByText } = render(<Notice/>);
        await waitFor(() => getByText('Notice'));
        await waitFor(() => getByText('2021.03.04'));
        await waitFor(() => getByText(/TmaxOS 21 업데이트/i));
        debug();
    });
});