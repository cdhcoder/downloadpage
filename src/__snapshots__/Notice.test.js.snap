// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notice/> matches snapshot 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
